### DISCUSSION TITLE
How can I disable caching for Incremental Static Regeneration in dev env?

### DISCUSSION THREAD

**Discussion by alexeysarakuz** [vercel/next.js#23917] [Help]
Hi guys, I'm using **ISR**, I have 2 environments (**dev & prod**) on Vercel.
In the dev environment, I want Nextjs to execute the **getStaticProps** each time I refresh the page (but it saves json files for each page and uses them in the future). I understand that this is the point of ISR but how can I disable it **only** for the dev environment?

**Comment by akellbl4** [comment]
Hi @alexeysarakuz
I have an idea of how to solve it. You can specify a time for ISR 1 second by default. Then depends on environment var `VERCEL_ENV`, change it to the time that you need.

For example, you will set a minimum time to revalidate on the build stage. When the app is running `revalidate` changes to the time that you need for a specific environment.
```js
const revalidateMap = {
  default: 1,
  production: 60,
  development: 1,
  preview: 60,
}

function getRevalidate(env) {
  if (!env || !revalidateMap[env]) {
    return revalidateMap.default
  }
  return revalidateMap[env]
}
export function getStaticPaths() {
  return {
    revalidate: getRevalidate(process.env.VERCEL_ENV)
  }
}
```

It's not disabling revalidation, but it should work.

**Reply by alexeysarakuz** [reply]
Thank you! 👍

**Reply by akellbl4** [reply]
@alexeysarakuz could you please mark the dissacussion as answered if it works for you :)

**Reply by alexeysarakuz** [reply]
@akellbl4 I am looking for how to disable it at all, if no one will answer I'll mark this as the answer

**Comment by brandonchinn178** [comment]
👍 It would be great to globally disable ISR in the application, without needing to do `revalidate: DISABLE_ISR ? 1 : 60` on every page.

I opened an issue here: https://github.com/vercel/next.js/issues/42186
